### PR TITLE
feat: add new event for lists and trigger it on save

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-1198

# Goal 

- When a remote list os local list gets updated we need to reindex all the resources that are using properties from this list

# TODO 

- [x] Unit tests
- [x] Dynamically retrieve the persistences

### CODE REVIEW

File: models/classes/search/tasks/UpdateResourceInIndex.php [hunk 1]
```diff
@@ -47,55 +45,82 @@ class UpdateResourceInIndex implements Action, ServiceLocatorAwareInterface, Tas
...
+    }
+
+    private function getSearchProxy(): SearchProxy
+    {
+        return $this->getServiceLocator()->get(SearchProxy::SERVICE_ID);
+    }
+
+    private function getIndexService(): IndexService
+    {
+        return $this->getServiceLocator()->get(IndexService::SERVICE_ID);
```
Comment (shpran) [inline models/classes/search/tasks/UpdateResourceInIndex.php:124]: Is it possible to use DI container here?

**Review by shpran — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: models/classes/search/tasks/UpdateResourceInIndex.php (hunk 1 above)
Comment (gabrielfs7) [reply models/classes/search/tasks/UpdateResourceInIndex.php:124]: doing, thanks!

**Review by hectoras — APPROVED** [review]
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Target branch is correct
- [x] Commits are following conventional commits
- [ ] New code is covered by tests (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Kitchen links are not exposed in the description

**Pull request merged**